src/interrupt.ts

```diff
@@ -0,0 +1,113 @@
+
+export function createInterruptHook(
+  toolConfigs: ToolInterruptConfig,
+  messagePrefix: string = "Tool execution requires approval",
+): (state: DeepAgentStateType) => Promise<Partial<DeepAgentStateType> | void> {
+  /**
+   * Create a post model hook that handles interrupts using native LangGraph schemas.
+   *
+   * Args:
+   *   toolConfigs: Record mapping tool names to HumanInterruptConfig objects
+   *   messagePrefix: Optional message prefix for interrupt descriptions
+   */
+
+  return async function interruptHook(
+    state: DeepAgentStateType,
+  ): Promise<Partial<DeepAgentStateType> | void> {
+    const messages = state.messages || [];
+    if (!messages.length) {
+      return;
+    }
+
+    const lastMessage = messages[messages.length - 1];
+    if (
+      !isAIMessage(lastMessage) ||
+      !lastMessage.tool_calls ||
+      !lastMessage.tool_calls.length
+    ) {
+      return;
+    }
+
+    // Separate tool calls that need interrupts from those that don't
+    const interruptToolCalls: ToolCall[] = [];
+    const autoApprovedToolCalls: ToolCall[] = [];
+
+    for (const toolCall of lastMessage.tool_calls) {
+      const toolName = toolCall.name;
+      if (toolName in toolConfigs) {
+        interruptToolCalls.push(toolCall);
+      } else {
+        autoApprovedToolCalls.push(toolCall);
+      }
+    }
+
+    if (!interruptToolCalls.length) {
+      return;
+    }
+
+    const approvedToolCalls = [...autoApprovedToolCalls];
+
+    const requests: HumanInterrupt[] = [];
+
+    for (const toolCall of interruptToolCalls) {
+      const toolName = toolCall.name;
+      const toolArgs = toolCall.args;
+      const description = `${messagePrefix}\n\nTool: ${toolName}\nArgs: ${JSON.stringify(toolArgs, null, 2)}`;
+      const toolConfig = toolConfigs[toolName];
+
+      const request: HumanInterrupt = {
+        action_request: {
+          action: toolName,
+          args: toolArgs,
+        },
+        config: toolConfig,
+        description: description,
+      };
+      requests.push(request);
+    }
+
+    const responses = await interrupt<HumanInterrupt[], HumanResponse[]>(
+      requests,
+    );
+
+    for (let i = 0; i < responses.length; i++) {
+      const response = responses[i];
+      const toolCall = interruptToolCalls[i];
+
+      if (response.type === "accept") {
+        approvedToolCalls.push(toolCall);
+      } else if (response.type === "edit" && response.args) {
+        const edited = response.args as ActionRequest;
+        const newToolCall = {
+          name: toolCall.name,
+          args: edited.args,
+          id: toolCall.id,
+        };
+        approvedToolCalls.push(newToolCall);
+      } else if (response.type === "ignore") {
+        continue;
+      } else if (response.type === "response") {
+        // continue here since it's a direct feedback
+        continue;
+      } else {
+        throw new Error(`Unknown response type: ${response.type}`);
+      }
```

Comment:
discussed in slack, lets throw err here and say it's not supported
```suggestion
      } else {
        throw new Error(`Unknown response type: ${response.type}`);
      }
```